core/src/main/java/io/grpc/PickFirstBalancerFactory.java

```diff
@@ -70,7 +72,7 @@ public void handleResolvedAddressGroups(
       EquivalentAddressGroup newEag = flattenEquivalentAddressGroup(servers);
       if (subchannel == null) {
         subchannel = helper.createSubchannel(newEag, Attributes.EMPTY);
-        helper.updatePicker(new Picker(PickResult.withSubchannel(subchannel)));
+        helper.updateBalancingState(CONNECTING, new Picker(PickResult.withSubchannel(subchannel)));
```

Comment:
The initial state of `ManagedChannelImpl` is `IDLE`. I think the method `handleResolvedAddressGroups()` wouldn't happen before `NameResolver.start()`, which wouldn't happen before `ManagedChannelImpl.exitIdle()`. So it should not be `IDLE` when this method is called.